common/sagas/wallet/wallet.ts

```diff
@@ -0,0 +1,250 @@
+import { showNotification } from 'actions/notifications';
+import {
+  setBalanceFullfilled,
+  setBalancePending,
+  setBalanceRejected,
+  setTokenBalancesPending,
+  setTokenBalancesFulfilled,
+  setTokenBalancesRejected,
+  setWallet,
+  setWalletConfig,
+  UnlockKeystoreAction,
+  UnlockMnemonicAction,
+  UnlockPrivateKeyAction,
+  ScanWalletForTokensAction,
+  SetWalletTokensAction,
+  TypeKeys
+} from 'actions/wallet';
+import { Wei } from 'libs/units';
+import { changeNodeIntent, web3UnsetNode, TypeKeys as ConfigTypeKeys } from 'actions/config';
+import { AddCustomTokenAction, TypeKeys as CustomTokenTypeKeys } from 'actions/customTokens';
+import { INode } from 'libs/nodes/INode';
+import {
+  IWallet,
+  MnemonicWallet,
+  getPrivKeyWallet,
+  getKeystoreWallet,
+  Web3Wallet,
+  WalletConfig
+} from 'libs/wallet';
+import { NODES, initWeb3Node } from 'config/data';
+import { SagaIterator } from 'redux-saga';
+import { apply, call, fork, put, select, takeEvery, take } from 'redux-saga/effects';
+import { getNodeLib } from 'selectors/config';
+import { getTokens, getWalletInst, getWalletConfig } from 'selectors/wallet';
+import { getCustomTokens } from 'selectors/customTokens';
+import translate from 'translations';
+import Web3Node, { isWeb3Node } from 'libs/nodes/web3';
+import { loadWalletConfig, saveWalletConfig } from 'utils/localStorage';
+import { getTokenBalances } from './helpers';
+
+export function* updateAccountBalance(): SagaIterator {
+  try {
+    yield put(setBalancePending());
+    const wallet: null | IWallet = yield select(getWalletInst);
+    if (!wallet) {
+      return;
+    }
+    const node: INode = yield select(getNodeLib);
+    const address = yield apply(wallet, wallet.getAddressString);
+    // network request
+    const balance: Wei = yield apply(node, node.getBalance, [address]);
+    yield put(setBalanceFullfilled(balance));
+  } catch (error) {
+    yield put(setBalanceRejected());
+  }
+}
+
+export function* updateTokenBalances(): SagaIterator {
+  try {
+    const wallet: null | IWallet = yield select(getWalletInst);
+    const config: null | WalletConfig = yield select(getWalletConfig);
+    if (!wallet || !config || !config.tokens) {
+      return;
+    }
+
+    // Get the token objects for the tokens in the config
+    const allTokens = yield select(getTokens);
+    const tokens = config.tokens
+      .map(symbol => allTokens.find(t => t.symbol === symbol))
+      .filter(token => !!token);
```

Comment:
Is the boolean typecast needed?

Reply:
a selector might be better in this case too

Reply:
Ah I thought boolean was important, the docs are inconsistent. At first it says:
> Function is a predicate, to test each element of the array. Return true to keep the element, false otherwise
But then it says:
> ...and constructs a new array of all the values for which callback returns a value that coerces to true.
so nope, not necessary!